packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ProjectResource.java

```diff
@@ -80,7 +79,8 @@ public Response createAsset(
 		@PathParam("resource_type") final String type, // ResourceType
 		@PathParam("resource_id") final String resourceId
 	) {
-		return proxy.createAsset(projectId, type, resourceId);
+		// String#toUpperCase is being used here to correct for Enum#valueOf usage, which we can't override
+		return proxy.createAsset(projectId, type.toUpperCase(), resourceId);
```

Comment:
this assumes that the value of the enum is just the lower case of the key. This is not very robust. I would suggest adding a look up function to the enum like:

```
public static ResourceType findByType(final String type) {
		return Arrays.stream(values()).filter(
			value -> value.type.equals(type)).findFirst().orElseThrow(() -> new IllegalArgumentException("No ResourceType with type: " + type)
		);
	}
```

and then using something like `ResourceType.findByType(type).type`